js/index.js

```diff
@@ -26,30 +43,18 @@ const loadData = () => {
 };
 const saveData = () => {
   add.addEventListener('click', () => {
-    const num = books.length + 1;
-    const book = { id: num, title: title.value, author: author.value };
-    books.push(book);
-    localStorage.setItem('books', JSON.stringify(books));
-    displayData(books);
-  });
-};
-const remove = () =>{
-  const remove = document.querySelectorAll('.remove');
-  remove.forEach(function (element) {
-    let id = parseInt(element.id);
-    element.addEventListener('click',() => {
-      let newbooks = books.filter (function (book) {
-        console.log('this is the id ',book.id);
-        if (book.id !== id)
-        return book;
-      })
-      books = newbooks;
+    if (title.value.length !== 0 && author.value.length !== 0) {
+      const num = books.length + 1;
+      const book = { id: num, title: title.value, author: author.value };
```

Comment:
- [OPTIONAL] Let me add some more suggestions to your codebase, please.

  - There is no need for testing the `.length` property of `title` and `author` values if were are only interested in checking whether there is input or not. Simply check for the truthiness of `value` being in the input field or not i.e. `(title.value && author.value)`.
  - When you assign the `num` variable the value of `books.length + 1`, you are setting the stage for invalid data manipulation. Let me give a scenario.

    If I have four books, then I delete the first one, the books remaining in my database will be books with `id`s `2,3 and 4`. Now the length of the books array will be `3`. What happens when I add another after this? I will have books with `id`s `2,3,4 and 4`. See evidence below:
    ![assign_id](https://user-images.githubusercontent.com/45002654/134045588-891e0c69-f5ce-49c0-aa97-f24fe30ce142.PNG)
    
    Now I have two books sharing the `id` number `4`.
    
    ## The problem
    When I delete any of the books with `id` of `4`, all of them get removed because we are filtering with id.
    
    ## Solution
    ```suggestion
        if (title.value && author.value) {
          const id = books.length ? books[books.length - 1].id + 1 : 1;
          const book = { id, title: title.value, author: author.value };
    ```
    If the books array has length(truthy), the new book added will have an `id` of `last book's id + 1`, else the new book will always start from one.